packages/core/src/codewhispererChat/tools/fsRead.ts

```diff
@@ -0,0 +1,131 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as vscode from 'vscode'
+import { getLogger } from '../../shared/logger/logger'
+import { readDirectoryRecursively } from '../../shared/utilities/workspaceUtils'
+
+const maxToolResponseSize = 30720 // 30KB
+
+enum OutputKind {
+    Text = 'text',
+    Json = 'json',
+}
+
+export interface InvokeOutput {
+    output: {
+        kind: OutputKind
+        content: string
+    }
+}
+
+export interface FsReadParams {
+    path: string
+    readRange?: number[]
+}
+
+export class FsRead {
+    private readonly fsPath: string
+    private readonly readRange?: number[]
+
+    constructor(params: FsReadParams) {
+        this.fsPath = params.path
+        this.readRange = params.readRange
+    }
+
+    public async invoke(): Promise<InvokeOutput> {
+        try {
+            const fileUri = vscode.Uri.file(this.fsPath)
+            let fileStat: vscode.FileStat
+            try {
+                fileStat = await vscode.workspace.fs.stat(fileUri)
+            } catch (err) {
+                throw new Error(`Path: "${this.fsPath}" does not exist or cannot be accessed. (${err})`)
+            }
+
+            if (fileStat.type === vscode.FileType.File) {
+                const fileContents = await this.readFile(fileUri)
+                getLogger().info(`Read file: ${this.fsPath}, size: ${fileContents.length}`)
+                return this.handleFileRange(fileContents)
+            } else if (fileStat.type === vscode.FileType.Directory) {
+                const maxDepth = this.getDirectoryDepth() ?? 0
+                const listing = await readDirectoryRecursively(fileUri, maxDepth)
+                return this.makeOutput(listing.join('\n'))
+            } else {
+                throw new Error(`"${this.fsPath}" is neither a standard file nor directory.`)
+            }
+        } catch (error: any) {
+            getLogger().error(`[fs_read] Failed to read "${this.fsPath}": ${error.message || error}`)
```

Comment:
You can actually initialize a logger in the constructor and pass in a topic (e.g. fsRead) and then use that instead. e.g. instead of having getLogger() everywhere and having to add a prefix of `[fs_read]` you can do:
```
this.logger = getLogger('fsRead')
...
this.logger.info('failed to read x,y,z')
```
and it will log: `fsRead: failed to read x,y,z` for you. I recommend you do that for all of the logs in this class so that you can easily associate what log messages are tied to what class

Reply:
Fixed.